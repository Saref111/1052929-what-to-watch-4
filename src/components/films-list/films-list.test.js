import React from "react";
import renderer from "react-test-renderer";
import FilmsList from "./films-list.jsx";
import {BrowserRouter, Route} from "react-router-dom";

const details = {
  bgPoster: `BG_HREF`,
  cover: `POSTER_HREF`,
  genre: `comedy`,
  year: 2001,
  rate: 8.8,
  votes: 666,
  director: `Kevin Smith`,
  actors: [`actor1`, `actor2`],
  description: {
    prescription: `prescription`,
    postscription: `postscription`,
  },
};

const FILMS = [
  {
    id: 0,
    title: `title`,
    src: `src`,
    preview: `preview`,
    details,
  },
  {
    id: 0,
    title: `title`,
    src: `src`,
    preview: `preview`,
    details,
  },
  {
    id: 0,
    title: `title`,
    src: `src`,
    preview: `preview`,
    details,
  },
  {
    id: 0,
    title: `title`,
    src: `src`,
    preview: `preview`,
    details,
  },
  {
    id: 0,
    title: `title`,
    src: `src`,
    preview: `preview`,
    details,
  },
  {
    id: 0,
    title: `title`,
    src: `src`,
    preview: `preview`,
    details,
  },
  {
    id: 0,
    title: `title`,
    src: `src`,
    preview: `preview`,
    details,
  },
  {
    id: 0,
    title: `title`,
    src: `src`,
    preview: `preview`,
    details,
  },
  {
    id: 0,
    title: `title`,
    src: `src`,
    preview: `preview`,
    details,
  },
];

describe(`Testing FilmsList component`, () => {
  test(`Snapshot test`, () => {
    const onHeaderClickHandler = jest.fn();
    const tree = renderer.create(<BrowserRouter><Route render={(props) => {
      return <FilmsList
        {...props}
        films={FILMS}
        onHeaderClickHandler={onHeaderClickHandler}
      />;
    }}/></BrowserRouter>

    );

    expect(tree).toMatchSnapshot();
  });
});
