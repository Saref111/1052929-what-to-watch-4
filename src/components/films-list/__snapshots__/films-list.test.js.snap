// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FilmsList component Snapshot test 1`] = `
Array [
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="title"
          height="175"
          src="src"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/0"
          onClick={[Function]}
        >
          title
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="title"
          height="175"
          src="src"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/0"
          onClick={[Function]}
        >
          title
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="title"
          height="175"
          src="src"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/0"
          onClick={[Function]}
        >
          title
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="title"
          height="175"
          src="src"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/0"
          onClick={[Function]}
        >
          title
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="title"
          height="175"
          src="src"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/0"
          onClick={[Function]}
        >
          title
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="title"
          height="175"
          src="src"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/0"
          onClick={[Function]}
        >
          title
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="title"
          height="175"
          src="src"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/0"
          onClick={[Function]}
        >
          title
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="title"
          height="175"
          src="src"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/0"
          onClick={[Function]}
        >
          title
        </a>
      </h3>
    </article>
  </div>,
  <div
    className="catalog__more"
  >
    <button
      className="catalog__button"
      onClick={[Function]}
      type="button"
    >
      Show more
    </button>
  </div>,
]
`;
